insecure/corruptible/network_common_test.go

```diff
@@ -0,0 +1,147 @@
+package corruptible
+
+import (
+	"fmt"
+	"github.com/onflow/flow-go/insecure"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/libp2p/message"
+	"github.com/onflow/flow-go/network/channels"
+	"github.com/onflow/flow-go/network/mocknetwork"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+)
+
+// This test file covers corruptible network tests that are not ingress or egress specific, including error conditions.
+
+// TestEngineClosingChannel evaluates that corruptible network closes the channel whenever the corresponding
+// engine of that channel attempts on closing it.
+func TestEngineClosingChannel(t *testing.T) {
+	corruptibleNetwork, adapter := corruptibleNetworkFixture(t, unittest.Logger())
+	channel := channels.TestNetworkChannel
+
+	// on invoking adapter.UnRegisterChannel(channel), it must return a nil, which means
+	// that the channel has been unregistered by the adapter successfully.
+	adapter.On("UnRegisterChannel", channel).Return(nil).Once()
+
+	err := corruptibleNetwork.EngineClosingChannel(channel)
+	require.NoError(t, err)
+
+	// adapter's UnRegisterChannel method must be called once.
+	mock.AssertExpectationsForObjects(t, adapter)
+}
+
+// TestProcessAttackerMessage_EmptyEgressIngressMessage checks that corruptible network returns an error
+// and exits if both egress and ingress messages are nil.
+func TestProcessAttackerMessage_EmptyEgressIngressMessage_Exit(t *testing.T) {
+	f := func(t *testing.T) {
+		ProcessAttackerMessage_EmptyEgressIngressMessage_Exit(t)
+	}
+	unittest.CrashTest(t, f, "both ingress and egress messages can't be nil", "TestProcessAttackerMessage_EmptyEgressIngressMessage_Exit")
+}
+
+func ProcessAttackerMessage_EmptyEgressIngressMessage_Exit(t *testing.T) {
```

Comment:
1. was addressed [here](https://github.com/onflow/flow-go/pull/2811/commits/a77e9335355b49a72a5303243a8c9b686ce0ac87#diff-7f9eb771807a19a9e6c927c72e57fed6f1f69d2ee400ab2216a15f621c1fb08a)
2. I added the snake case to differentiate what we were testing for each test of `ProcessAttackMessage()` - there are 2 error conditions: a) both ingress and egress are nil b) both ingress and egress are not nil and I wanted to make that clearer. If it was all camel case, it would be less readable because the function name needs to have 2 things: a) name of the function we're testing b) specific condition we're testing of that function

Reply:
3. addressed in https://github.com/onflow/flow-go/pull/2811/commits/08ba897a2cff40ac67ca49399fc2464f26dca186 I took out the extra code that creates a message fixtures since we're sending an empty message.

Reply:
4. addressed in https://github.com/onflow/flow-go/pull/2811/commits/08ba897a2cff40ac67ca49399fc2464f26dca186 I couldn't remove the WaitGroup because that would cause a race condition and the corrupted network would be cancelled before the test completes. So it was necessary to leave in the WaitGroup.